src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -0,0 +1,310 @@
+import React from 'react'
+import {SearchIcon, XCircleFillIcon, XIcon} from '@primer/octicons-react'
+import {FocusKeys} from '@primer/behaviors'
+
+import {
+  Button,
+  IconButton,
+  Heading,
+  Box,
+  AnchoredOverlay,
+  Tooltip,
+  TextInput,
+  AnchoredOverlayProps,
+  ActionList,
+  Spinner,
+  Text,
+} from '../../../src/index'
+import {useSlots} from '../../hooks/useSlots'
+import {ClearIcon} from './tmp-ClearIcon'
+
+const SelectPanelContext = React.createContext({
+  onCancel: () => {},
+  onClearSelection: () => {},
+  searchQuery: '',
+  setSearchQuery: () => {},
+})
+
+// @ts-ignore todo
+const SelectPanel = props => {
+  const anchorRef = React.useRef<HTMLButtonElement>(null)
+
+  // 🚨 Hack for good API!
+  // we strip out Anchor from children and pass it to AnchoredOverlay to render
+  // with additional props for accessibility
+  let renderAnchor: AnchoredOverlayProps['renderAnchor'] = null
+  const contents = React.Children.map(props.children, child => {
+    if (child.type === SelectPanelButton) {
+      renderAnchor = anchorProps => React.cloneElement(child, anchorProps)
+      return null
+    }
+    return child
+  })
+
+  // TODO: defaultOpen is not for debugging, I don't intend to make
+  // it part of the API
+  const [open, setOpen] = React.useState(props.defaultOpen)
+
+  const onInternalClose = () => {
+    setOpen(false)
+
+    if (typeof props.onCancel === 'function') props.onCancel()
+  }
+  // @ts-ignore todo
+  const onInternalSubmit = event => {
+    setOpen(false)
+    if (typeof props.onSubmit === 'function') props.onSubmit(event)
+  }
+
+  const onInternalClearSelection = () => {
+    if (typeof props.onSubmit === 'function') props.onClearSelection()
+  }
+
+  /* Search/Filter */
+  const [searchQuery, setSearchQuery] = React.useState('')
+
+  return (
+    <>
+      <AnchoredOverlay
+        anchorRef={anchorRef}
+        renderAnchor={renderAnchor}
+        open={open}
+        onOpen={() => setOpen(true)}
+        onClose={onInternalClose}
+        width="medium"
+        height="large"
+        focusZoneSettings={{bindKeys: FocusKeys.Tab}}
+      >
+        {/* TODO: Keyboard navigation of actionlist should be arrow keys
+            with tabs to enter and escape
+        */}
+        <SelectPanelContext.Provider
+          value={{
+            onCancel: onInternalClose,
+            onClearSelection: onInternalClearSelection,
+            searchQuery,
+            // @ts-ignore todo
+            setSearchQuery,
+          }}
```

Comment:
How many subscribers do we think will hook into this context value? Will it ever be worth eventually finding a way to create a stable value or split up the context into readers/writers or are we thinking there aren't too many updates or readers here? 👀